File: order-service/src/main/java/com/avenga/orderservice/model/persistence/product/Product.java
package com.avenga.orderservice.model.persistence.product;

import lombok.*;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Product {
    private Long id;

    private String name;

    private String description;

    private Integer price;

    private boolean available;

    private int availableProductCount;
}

File: frontend/src/app/modules/products/products-from-database/interface/product.ts
import {OrderStatus} from "../../../orders/orders-from-database/interface/order";

export interface Product {
  id: number;
  name: string;
  description: string;
  price: number;
  available: boolean;
  availableProductCount: number;
}

export interface NewProduct {
  name: string;
  description: string;
  price: number;
  available: boolean;
  availableProductCount: number;
}
